src/common/literals.cpp

```diff
@@ -11,16 +11,18 @@
 #include <libp2p/multi/multihash.hpp>
 #include <libp2p/peer/peer_id.hpp>
 
+#include <algorithm>
+
 namespace libp2p::common {
   libp2p::common::Hash256 operator""_hash256(const char *c, size_t s) {
     libp2p::common::Hash256 hash{};
-    std::copy_n(c, std::min(s, 32ul), hash.rbegin());
+    std::copy_n(c, std::min(s, static_cast<size_t>(32ul)), hash.rbegin());
```

Comment:
just a minor note.
It looks like `l` type specifier for constants can be omitted due to explicit `static_cast`.

Reply:
Should I drop the `u` type specifier too?

Reply:
`u` could still be kept as an indicator of the value's "nature".
That is not an issue at all. We may leave it as is even with `l`.

Reply:
I removed the `l` and left the `u`. this leaves the indicator of its "nature", and de-duplicates the notion of its "size".

Reply:
imho, casting to the same type explicitly looks more clear.